### PR TITLE
Adding ExecutionEngine::add_global_mapping

### PR DESCRIPTION
**Pull request by 6A**
It works on my machine (and some tests have been made related to it), so it should be good. I'll add `no_run` if Travis fails to build this.

### CODE REVIEW

File: src/execution_engine.rs [hunk 1]
```diff
@@ -51,11 +51,50 @@ impl ExecutionEngine {
...
 
-    // pub fn add_global_mapping(&mut self, global: &AnyValue, addr: *const ()) {
-    //     unsafe {
-    //         LLVMAddGlobalMapping(self.execution_engine, global.as_value_ref(), addr as *mut ::libc::c_void)
-    //     }
-    // }
+    /// Maps the specified value to an address.
+    ///
+    /// # Example
+    /// ```
```
Comment (TheDan64) [inline src/execution_engine.rs]: Likely going to need no_run

File: src/execution_engine.rs [hunk 2]
```diff
@@ -51,11 +51,50 @@ impl ExecutionEngine {
...
+    /// builder.build_return(Some(&retv));
+    ///
+    /// let mut ee = module.create_jit_execution_engine(OptimizationLevel::None).unwrap();
+    /// ee.add_global_mapping(&extf, sumf as *mut _);
+    ///
+    /// let result = unsafe { ee.run_function(&f, &[]) }.as_float(&ft);
+    ///
+    /// assert_eq!(result, 128.);
+    /// ```
+    pub fn add_global_mapping(&mut self, value: &AnyValue, addr: *const ()) {
```
Comment (TheDan64) [inline src/execution_engine.rs]: I still think addr should be a `usize`, since it's safe to cast to a usize but not a `*const ()`?

**Comment by 6A** [comment]
Huh, gonna have to fix this. Do note that the [beta build](https://travis-ci.org/TheDan64/inkwell/jobs/286703862) actually passed, whereas the [stable](https://travis-ci.org/TheDan64/inkwell/jobs/286703861) and [nightly](https://travis-ci.org/TheDan64/inkwell/jobs/286703863) ones didn't.

This kind of "random" bug adds up to what I previously said on #13.

**Comment by TheDan64** [comment]
Yeah, it's basically pure chance as to whether or not travis decides rustdoc is taking too long and kills it

File: src/execution_engine.rs (hunk 2 above)
Comment (6A) [reply src/execution_engine.rs]: Alright, switched to `usize`.

**Comment by 6A** [comment]
Once again a fail in something completely different that I haven't even touched...

**Comment by TheDan64** [comment]
I restarted the job and it passed that time.

File: examples/kaleidoscope/main.rs [hunk 3]
```diff
@@ -1193,16 +1193,15 @@ macro_rules! print_flush {
 // However, extern functions such as cos(x) and sin(x) can be imported without any problem.
 // Other lines related to this program can be found further down.
 
-// pub extern "C" fn putchard(x: f64) -> f64 {
-//     print_flush!("{}", x as u8 as char);
-//     x
-// }
+pub extern fn putchard(x: f64) -> f64 {
```
Comment (TheDan64) [inline examples/kaleidoscope/main.rs]: These two extern functions are no longer used.. did you mean to remove them?

Comment (6A) [reply examples/kaleidoscope/main.rs]: Ugh, I tried to use them from the Kaleidoscope REPL, but didn't succeed, and forgot to comment them once again.

**Pull request merged**